app/controllers/meetings_controller.rb

```diff
@@ -45,58 +26,40 @@ def comment
 
   # rubocop:disable MethodLength
   def delete_comment
-    comment_exists = Comment.where(id: params[:commentid]).exists?
-    is_my_comment = Comment.where(
-      id: params[:commentid],
-      comment_by: current_user.id
-    ).exists?
+    comment_exists = Comment.exists?(id: params[:commentid])
+    is_my_comment = Comment.user_comment?(params[:commentid], current_user)
 
     if comment_exists
-      meeting_id = Comment.where(id: params[:commentid]).first.commentable_id
-      is_my_meeting = MeetingMember.where(
-        meeting_id: meeting_id,
-        user_id: current_user.id,
-        leader: true
-      ).exists?
-      is_member = MeetingMember.where(
-        meeting_id: meeting_id,
-        user_id: current_user.id
-      ).exists?
-    else
-      is_my_meeting = false
-      is_member = false
+      meeting_id = Comment.find(params[:commentid]).commentable_id
+      meeting = Meeting.find(meeting_id)
+      is_my_meeting = MeetingMember.leader?(current_user, meeting)
```

Comment:
We should be asking the `MeetingMember` object about its state and not the class. At the top of my head, I would do something like:
```ruby
meeting_member = meeting.members.find_by(user: current_user)
```
Chaining methods like this ensures that the record returned is a member of the meeting. Just refactor if gets too long.

Then I would add the methods `#leader?` to the `MeetingMember` model.

Reply:
I am trying to convert 

```ruby
MeetingMember.find_by(
        id: current_user.id, leader: true
      )
```
into the above format but it fails because the result of `meeting.members.find_by(current_use.idr)` is`User` not `MeetingMember` - so determining if it's leader or not might be a little bit hard